velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -522,8 +538,24 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
   return nullptr;
 }
 
+const TypePtr& ReaderBase::getRequestedType(
+    const TypePtr& requestedType,
+    std::string& name) const {
+  if (!requestedType || !requestedType->isRow()) {
+    return requestedType;
+  }
+
+  const auto& rowType = requestedType->asRow();
+  if (!rowType.containsChild(name)) {
+    return requestedType;
```

Comment:
This does not look quite right.  Shall we just throw exception if the requested name is not in requested schema?